easier68k/core/opcodes/jsr.py

```diff
@@ -0,0 +1,256 @@
+        :param command: The command string to check (e.g. 'MOVE.B', 'LEA', etc.)
+        :return: Whether the string is an instance of this command type
+        """
+        return opcode_util.command_matches(command, 'JSR')
+
+    @classmethod
+    def get_word_length(cls, command: str, parameters: str) -> int:
+        """
+        >>> Jsr.get_word_length('JSR', '(A5)')
+        1
+
+        >>> Jsr.get_word_length('JSR', '-(A7)')
+        1
+
+        >>> Jsr.get_word_length('JSR', '($BBBB).W')
+        2
+
+        >>> Jsr.get_word_length('JSR', '($1000).W')
+        2
+
+        >>> Jsr.get_word_length('JSR', '($FFFF).L')
+        3
+
+        >>> Jsr.get_word_length('JSR', '($8000).L')
+        3
+
+        Gets what the end length of this command will be in memory
+        :param command: The text of the command itself (e.g. "LEA", "MOVE.B", etc.)
+        :param parameters: The parameters after the command
+        :return: The length of the bytes in memory in words, as well as a list of warnings or errors encountered
+        """
+
+        dest = parse_assembly_parameter(parameters.strip())  # Parse the destination
+
+        length = 1  # Always 1 word not counting additions to end
+
+        if dest.mode == EAMode.AWA:  # Appends a word
+            length += 1
+
+        if dest.mode == EAMode.ALA:  # Appends a long, so 2 words
+            length += 2
+
+        return length
+
+    @classmethod
+    def is_valid(cls, command: str, parameters: str) -> (bool, list):
+        """
+        Tests whether the given command is valid
+
+        >>> Jsr.is_valid('JSR', '(A0)')[0]
+        True
+
+        >>> Jsr.is_valid('JSR', '(A5)')[0]
+        True
+
+        >>> Jsr.is_valid('JSR', '#5, D1')[0]
+        False
+
+        >>> Jsr.is_valid('JSR.W', '(A1)')[0]
+        True
+
+        >>> Jsr.is_valid('JS', '($1000).W')[0]
+        False
+
+        >>> Jsr.is_valid('JSR.', '(A4)')[0]
+        False
+
+        >>> Jsr.is_valid('JSR', '($7000).W')[0]
+        True
+
+        :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
+        :param parameters: The parameters after the command (such as the source and destination of a move)
+        :return: Whether the given command is valid and a list of issues/warnings encountered
+        """
+        return opcode_util.n_param_is_valid(command, parameters, "JSR", 1, param_invalid_modes=[[EAMode.DRD,
+                                                                                                 EAMode.ARD,
+                                                                                                 EAMode.ARIPD,
+                                                                                                 EAMode.ARIPI,
+                                                                                                 EAMode.IMM]])[:2]
+
+    @classmethod
+    def disassemble_instruction(cls, data: bytearray) -> Union[Opcode, None]:
```

Comment:
What's with the `Union` here? Doesn't that break the pattern followed by the other functions?

Reply:
Hmm, I think this makes it more clear as to the different return types you can get with this function (since it can either return an Opcode object or None, depending on the input).